Python/bytecodes.c

```diff
@@ -2359,16 +2372,35 @@ dummy_func(
             assert(Py_TYPE(owner_o)->tp_flags & Py_TPFLAGS_MANAGED_DICT);
             PyDictObject *dict = _PyObject_GetManagedDict(owner_o);
             DEOPT_IF(dict == NULL);
+            DEOPT_IF(!LOCK_OBJECT(dict));
+            #ifdef Py_GIL_DISABLED
+            dict = _PyObject_GetManagedDict(owner_o);
+            if (dict == NULL) {
```

Comment:
This may overwrite the local variable `dict` with a *different* dictionary, which will cause problems for subsequent calls like `UNLOCK_OBJECT(dict)`.

I'm not entirely sure if we need a check list this here, but if we do, I think it should be something like:

```
#ifdef Py_GIL_DISABLED
if (dict != _PyObject_GetManagedDict(owner_o)) {
    UNLOCK_OBJECT(dict);
    DEOPT_IF(true);
}
#endif
```